Forte.EpiServer.AzureSearch/Extensions/XhtmlStringExtensions.cs

```diff
@@ -1,22 +1,60 @@
+using System.Text;
 using AngleSharp;
 using AngleSharp.Dom;
+using EPiServer;
 using EPiServer.Core;
 using EPiServer.Core.Html;
+using EPiServer.Core.Html.StringParsing;
+using EPiServer.Security;
+using EPiServer.ServiceLocation;
 
 namespace Forte.EpiServer.AzureSearch.Extensions
 {
     public static class XhtmlStringExtensions
     {
         public static string GetPlainTextContent(this XhtmlString xhtmlString)
         {
-            var htmlWithoutScripts = RemoveScripts(xhtmlString.ToHtmlString());
-            //TODO: try to inline content of embeded blocks
-            return StripHtml(htmlWithoutScripts);
+            var textBuilder = new StringBuilder();
+            var contentLoader = ServiceLocator.Current.GetInstance<IContentLoader>();
+
+            var xhtmlFragments = xhtmlString
+                .Fragments
+                .GetFilteredFragments(PrincipalInfo.AnonymousPrincipal);
+            
+            foreach (var fragment in xhtmlFragments)
+            {
+                switch (fragment)
+                {
+                    case ContentFragment contentFragment when ContentReference.IsNullOrEmpty(contentFragment.ContentLink):
+                        continue;
+                    case ContentFragment contentFragment:
+                    {
+                        var content = contentLoader.Get<IContent>(contentFragment.ContentLink);
+                        
+                        var text = content.ExtractTextFromBlock();
+                        textBuilder.Append(text);
+                        break;
+                    }
+                    case StaticFragment staticFragment:
+                        var html = staticFragment.InternalFormat;
+                        var htmlWithoutScripts = RemoveScripts(html);
+                        
+                        const string whitespace = " ";
+                        var htmlWithTrailingSpace =
+                            htmlWithoutScripts.EndsWith(whitespace) ? htmlWithoutScripts : htmlWithoutScripts + whitespace;
```

Comment:
why do we need whitespace?

Reply:
I've adjusted the solution with one const for whitespace in BlockContentExtractorController and two places, where we use it for joining the individual extraction results for one document.ContentBody to display in the search results.